inflate_mmap.go

```
@@ -0,0 +1,74 @@
+	if pos < 0 || pos+2 > int64(len(data)) {
+		return nil, nil, io.ErrUnexpectedEOF
+	}
+
+	cmf, flg := data[pos], data[pos+1]
+	if cmf&0x0f != 8 || cmf>>4 > 7 || flg&0x20 != 0 || (uint16(cmf)<<8|uint16(flg))%31 != 0 {
+		return nil, nil, fmt.Errorf("invalid zlib header at pack offset %d", pos)
+	}
+
+	br := getBytesReader(data[pos+2:])
+	fr := flatePool.Get().(io.ReadCloser)
+	if err := fr.(flate.Resetter).Reset(br, nil); err != nil {
```

golangci-lint (2.12.2)

[error] right hand must be only type assertion

(forcetypeassert)